microovn/database/certificate.go

```diff
@@ -0,0 +1,33 @@
+package database
+
+//go:generate -command mapper lxd-generate db mapper -t certificate.mapper.go
+//go:generate mapper reset
+//
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e Certificate objects table=certificate
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e Certificate objects-by-Name table=certificate
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e Certificate id table=certificate
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e Certificate create table=certificate
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e Certificate delete-by-Name table=certificate
+//go:generate mapper stmt -d github.com/canonical/microcluster/cluster -e Certificate update table=certificate
+//
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e Certificate GetMany table=certificate
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e Certificate GetOne table=certificate
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e Certificate ID table=certificate
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e Certificate Exists table=certificate
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e Certificate Create table=certificate
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e Certificate DeleteOne-by-Name table=certificate
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e Certificate DeleteMany-by-Name table=certificate
+//go:generate mapper method -i -d github.com/canonical/microcluster/cluster -e Certificate Update table=certificate
+
+// Certificate is used to track the Certificates used by OVN.
+type Certificate struct {
+	ID   int
+	Name string `db:"primary=yes"`
```

Comment:
> The certificate contains a Serial number which uniquely identifies the certificate from this CA. Would it make sense to use that as a primary key instead?

Currently only the CA cert/key is stored in the database, certificates for ovn services are managed locally on disk. So given that the expectation is to have only one record in this table at the time, I don't think it adds too much value to store serial number as well.

> Do we need to store date of issue and expiry?

As mentioned above, non-CA certificates are not stored in the database so any tool that handles certificate lifecycle should not depend too much on the database either. I think we should rely on the database as little as possible because it'll make later integration with other certificates sources easier.

Reply:
Right, the description is misleading then. If this is only used for the CA-cert should it be named specifically for that as well?

Reply:
Oh, you mean the comment above structure, yes, that could be improved. Thanks.

Reply:
Well, not just the comment. If this database table and data structure is for use solemnly by CA Certificates, why give the impression that you are trying to implement a generic data structure for certificates?

Reply:
So, I want to preface this with the fact that I'm not supper happy about creating table that is expected to hold only single row of data. However since we need to share the CA cert and key between members I designed a generic table that can hold certificate/key pairs. Isn't it better to have generic table structure so that it's potentially useful in the future?  What would be your design for this table, because since CA cert differs from "regular" cert on such a low level I'm struggling to come up with a way to effectively distinguish it on DB table level.

However, the more I think about it, maybe the standalone table was bad approach after all. Maybe we should just use existing `config` table that accepts keys/values and store our CA as `ca_cert`, `ca_key` in the config database. What do you think about this approach?

Reply:
I quite like the idea of just stashing it in key/value pairs in the `config` table, go for it!

If we in the future get a need for a full blown data structure for multiple certificate types etc, we then probably also have specific use cases to feed the design work of it.